src/test/java/org/kainos/ea/service/UserServiceTest.java

```diff
@@ -251,6 +257,47 @@ public void generateHash_shouldNotReturnIdenticalHash_whenGivenDifferentInput()
         String first    = "IDENTICALSTRING";
         String second   = "NOTIDENTICALSTRING";
 
-        Assertions.assertNotEquals( userService.generateHash( first ), userService.generateHash( second ) );
+        assertNotEquals( userService.generateHash( first ), userService.generateHash( second ) );
+    }
+
+    @Test
+    public void registerUser_shouldReturnID_whenCredentialsAreValid() throws DatabaseConnectionException, SQLException {
+        int expectedResult = 1;
+        Mockito.when(databaseConnector.getConnection()).thenReturn(conn);
+        Mockito.when(userData.registerUser(userRequest, conn)).thenReturn(expectedResult);
+
+        int actualResult = userService.registerUser(userRequest);
+
+        assertEquals(expectedResult, actualResult);
+    }
+
+    @Test
+    void registerUser_shouldThrow400Error_whenUserCredentialsAreInvalid() throws DatabaseConnectionException, SQLException {
```

Comment:
I'm assuming your userData class does not return a 400 if user credentials are invalid as that would be a valid ID

Reply:
I'm not sure what you mean by this comment

Reply:
test removed